lib/diff/lcs/internals.rb

```diff
@@ -60,22 +59,18 @@ def lcs(a, b)
     end
 
     # Now, compute the equivalence classes of positions of elements.
+    # An explanation for how this works: https://codeforces.com/topic/92191/en7
     b_matches = position_hash(b, b_start..b_finish)
 
     thresh = []
     links  = []
-    string = a.kind_of?(String)
 
     (a_start..a_finish).each do |i|
-      ai = string ? a[i, 1] : a[i]
+      ai = a[i]
       bm = b_matches[ai]
       k = nil
       bm.reverse_each do |j|
-        if k and (thresh[k] > j) and (thresh[k - 1] < j)
-          thresh[k] = j
-        else
-          k = replace_next_larger(thresh, j, k)
-        end
+        k = replace_next_larger(thresh, j)
```

Comment:
From what I understand (from the link I added in line 62), this is enough for the algorithm to be correct, with this change I don't see any test failed, so I guess it's either true or we're missing some test.

Reply:
I’ll need to take some time to read the link and review and see if larger or smaller test data causes issues. It has been some years since I’ve looked at this and have mostly forgotten how it works (so I’m very _happy_ someone is looking at this).

Reply:
Yeah please take your time. Just wanna say that I really appreciate your work on this library, I have a good time reading it, learning some interesting algorithms.

Reply:
I think that you’re right on this. We will want to do two things so it’s easier to revert if this turns out to be incorrect.

1. We’re going to put out a prerelease version. I’ll use it to run the rspec unit tests.
2. I’d like to put the threshold map condition into a comment so that it’s easier to see for revert if it is problematic.

We only need to put the `if` clause in a comment, so:

```ruby
# Previously, we would update `thresh[k] = j` if `k and (thresh[k] > j) and (thresh[k - 1] < j)`
# Otherwise, we would do this. The update appears not to be necessary, so we are trying to
# simplify to:
k = replace_next_larger
```

Reply:
Yeah, I've added the comment